csrc/executor_utils.cpp

```diff
@@ -407,11 +407,67 @@ void validateAlignedVectorizeExtents(
   //     info.word_size);
 }
 
+namespace {
+
+// Return offsets of the first points accessed. Currently only
+// non-zero when tensor is sliced.
+std::vector<size_t> getTensorOffsets(
+    TensorView* tv,
+    const std::vector<int64_t>& strides,
+    ExpressionEvaluator& eval) {
+  if (!tv->isFusionInput()) {
+    return {0};
+  }
+
+  std::vector<size_t> offsets;
+  offsets.reserve(tv->uses().size());
+
+  for (auto use : tv->uses()) {
+    auto slice = dynamic_cast<SliceOp*>(use);
+
+    if (slice == nullptr) {
+      offsets.push_back(0);
+      continue;
+    }
+
+    // Since slice is done with the rfactor domain and we need to know
+    // their strides, the rfactor domain must be the same as the allocation
+    // domain
+    TORCH_INTERNAL_ASSERT(
+        tv->getMaybeAllocationDomain() == tv->getMaybeRFactorDomain(),
```

Comment:
Can we remove this check if we call this function with `metadata["logical_stride"]`?

Reply:
I think so. I just didn't remember we have `logical_stride` defined. Let me update the PR

Reply:
Updated